src/Lykke.Job.OpsGenie.Services/QueueReader/DomainQueueReaderFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading.Tasks;
+using Lykke.Common.Log;
+using Lykke.Job.OpsGenie.AzureRepositories.Queue;
+using Lykke.Job.OpsGenie.Contract;
+using Lykke.Job.OpsGenie.Core.Domain;
+using Lykke.Job.OpsGenie.Services.Mappers;
+using Lykke.SettingsReader;
+
+namespace Lykke.Job.OpsGenie.Services.QueueReader
+{
+    public class DomainQueueReaderFactory:IDomainQueueReaderFactory
+    {
+        private readonly IReloadingManager<string> _connString;
+        private readonly ILogFactory _logFactory;
+        private readonly IApiAdapterStorage _apiAdapterStorage;
+
+        public DomainQueueReaderFactory(IReloadingManager<string> connString, ILogFactory logFactory, IApiAdapterStorage apiAdapterStorage)
+        {
+            _connString = connString;
+            _logFactory = logFactory;
+            _apiAdapterStorage = apiAdapterStorage;
+        }
+
+        public IDomainQueueReader Create(string domain)
+        {
+            return new DomainQueueReader(_connString, domain, _logFactory, ProcessQueueMessage);
+        }
+
+        private async Task ProcessQueueMessage(AlertQueueMessage queueMessage)
+        {
+            var apiAdapter = _apiAdapterStorage.GetOrDefault(queueMessage.Domain);
+
+            try
+            {
+                await apiAdapter.RaiseAlertAsync(queueMessage.MapFromQueueMessage());
+            }
+            catch (Exception e)
+            {
+                _logFactory.CreateLog(apiAdapter).Error(e, context: queueMessage);
```

Comment:
1. Generally pattern is - create log in the ctor and use it the methods
2. And again, generally exceptions should be handled and logged at the top most level (by infrastructure). Will be good to add try/catch/log to the AzureQueueSubscriber and remove it from here